File: FlowRhApi/Controllers/BeneficioController.cs
using FlowRhApi.Dto.Beneficio;
using FlowRhApi.Models;
using FlowRhApi.Repositories.Beneficio;
using Microsoft.AspNetCore.Mvc;

namespace FlowRhApi.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class BeneficioController : ControllerBase
    {
        private readonly IBeneficioInterface _beneficioInterface;

        public BeneficioController(IBeneficioInterface BeneficioInterface)
        {
            _beneficioInterface = BeneficioInterface;
        }

        [HttpGet("ListarBeneficios")]
        public async Task<ActionResult<ResponseModel<List<BeneficioModel>>>> ListarBeneficio()
        {
            var beneficiosListReponse = await _beneficioInterface.ListarBeneficios();
            return Ok(beneficiosListReponse);
        }

        [HttpGet("BuscarBeneficioPorId")]
        public async Task<ActionResult<ResponseModel<BeneficioModel>>> BuscarBeneficioPorId(int idBeneficio)
        {
            var beneficiosListReponse = await _beneficioInterface.BuscarBeneficioPorId(idBeneficio);
            return Ok(beneficiosListReponse);
        }

        [HttpPost("CriarBeneficio")]
        public async Task<ActionResult<ResponseModel<List<BeneficioModel>>>> CriarBeneficio(BeneficioCriacaoDto beneficioCriacaoDto)
        {
            var beneficiosListReponse = await _beneficioInterface.CriarBeneficio(beneficioCriacaoDto);
            return Ok(beneficiosListReponse);
        }

        [HttpDelete("DeletarBeneficio/{idBeneficio}")]
        public async Task<ActionResult<ResponseModel<List<BeneficioModel>>>> DeletarBeneficio(int idBeneficio)
        {
            var beneficiosListReponse = await _beneficioInterface.DeletarBeneficio(idBeneficio);
            return Ok(beneficiosListReponse);
        }
        
        [HttpPut("EditarBeneficio")]
        public async Task<ActionResult<ResponseModel<List<BeneficioModel>>>> EditarBeneficio(BeneficioEdicaoDto BeneficioEdicaoDto)
        {
            var beneficiosListReponse = await _beneficioInterface.EditarBeneficio(BeneficioEdicaoDto);
            return Ok(beneficiosListReponse);
        }
    }
}

File: frontend/lib/domain/controllers/beneficio_controller.dart
import 'package:flow_rh/domain/dto/beneficio_criacao_dto.dart';
import 'package:flow_rh/domain/models/beneficios.dart';
import 'package:flow_rh/domain/models/response_model.dart';
import 'package:flow_rh/domain/repositories/beneficios_repository.dart';

class BeneficioController {
  final IBeneficiosRepository repository;

  BeneficioController(this.repository);

  Future<List<ResponseModel<Beneficio>>> ListarBeneficios() async {
    // Converte o ViewModel para JSON antes de enviar

    // Envia para o repositório
    final result = await repository.listarBeneficios();
    return result;
  }

  Future<List<ResponseModel<Beneficio>>> DeletarBeneficio(
      int idBeneficio) async {
    // Converte o ViewModel para JSON antes de enviar
    // Envia para o repositório
    final result = await repository.deletarBeneficio(idBeneficio);
    return result;
  }

  Future<List<ResponseModel<Beneficio>>> CriarBeneficio(
      BeneficioCriacaoDto beneficioCriacaoDto) async {
    // Converte o ViewModel para JSON antes de enviar
    // Envia para o repositório
    final result = await repository.criarBeneficio(beneficioCriacaoDto);
    return result;
  }

  Future<List<ResponseModel<Beneficio>>> AtualizarBeneficio(
      Beneficio beneficioEdicaoDto) async {
    // Converte o ViewModel para JSON antes de enviar
    // Envia para o repositório
    final result = await repository.editarBeneficio(beneficioEdicaoDto);
    return result;
  }
}
